Tools/TestWebKitAPI/Tests/WebKitCocoa/ResourceLoadStatistics.mm

```diff
@@ -2174,3 +2175,31 @@ HTTPServer httpServer({
     EXPECT_FALSE(didSendSite2CookieHeader);
     didSendSite2CookieHeader = false;
 }
+
+TEST(ResourceLoadStatistics, LoginStatusRequiresPasswordAutoFill)
+{
+    TestWebKitAPI::HTTPServer server({
+        { "/index"_s, { @"" } }
+    });
+    RetainPtr configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+
+    for (_WKFeature *feature in WKPreferences._features) {
+        if ([feature.key isEqualToString:@"LoginStatusAPIEnabled"] || [feature.key isEqualToString:@"LoginStatusAPIRequiresWebAuthnEnabled"])
+            [[configuration preferences] _setEnabled:YES forFeature:feature];
+    }
+    RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 100, 100) configuration:configuration.get()]);
+    [webView loadRequest:server.request("/index"_s)];
+    [webView _didFillPasswordForUsername:@"example"];
+
+    [webView evaluateJavaScript:@"navigator.setStatus('logged-in');" completionHandler:^(id, NSError *) {
+        [webView evaluateJavaScript:@"navigator.isLoggedIn()" completionHandler:^(id result, NSError *) {
+            EXPECT_TRUE([result boolValue]);
+        }];
+    }];
+
+    [webView evaluateJavaScript:@"navigator.setStatus('logged-out');" completionHandler:^(id, NSError *) {
+        [webView evaluateJavaScript:@"navigator.isLoggedIn()" completionHandler:^(id result, NSError *) {
+            EXPECT_FALSE([result boolValue]);
```

Comment:
This test needs to wait until the completion handlers are called before finishing. See the uses of `TestWebKitAPI::Util::run(&done);` in this file for an example.